gphotofs.c

```diff
@@ -871,10 +845,66 @@ gphotofs_connect()
                 gp_port_info_get_path (info, &xpath);
                 gp_setting_set("gphoto2", "port", xpath);
 
+                // According to the gphoto2 docs, both port and abilities must be set or the camera will be
+                // auto-detected.
+                if (!sModel) {
+                    CameraList *cameraList;
+                    ret = gp_list_new(&cameraList);
+                    if (ret != 0)
+                        break;
+
+                    ret = gp_abilities_list_detect(p->abilities, il, cameraList, p->context);
+                    if (ret != 0)
+                        break;
+
+                    ret = gp_list_count(cameraList);
+                    if (ret > 1) {
+
+                        g_fprintf(stderr, "Multiple cameras detected on specified port. Model is required.\n");
+                        return GP_ERROR;
```

Comment:
the list is never freed. Here or below.

Reply:
Is it necessary? I'm considering this to be the same case as ret != 0, where the whole program exits immediately without proceeding to the actual mount.